kratos/processes/compute_nodal_gradient_process.h

```diff
@@ -115,45 +155,66 @@ struct VariableVectorRetriever
  * @details This process computes the gradient of a certain variable stored in the nodes
  * @author Riccardo Rossi
  * @author Vicente Mataix Ferrandiz
- * @tparam THistorical If the variable is historical or not
+ * @tparam TOutputHistorical If the output variable is historical or not
 */
-template<bool THistorical>
+template<bool TOutputHistorical>
 class KRATOS_API(KRATOS_CORE) ComputeNodalGradientProcess
     : public Process
 {
 public:
     ///@name Type Definitions
     ///@{
 
-    /// The definition of the node
-    typedef Node NodeType;
-
     /// Pointer definition of ComputeNodalGradientProcess
     KRATOS_CLASS_POINTER_DEFINITION(ComputeNodalGradientProcess);
 
     ///@}
     ///@name Life Cycle
     ///@{
 
-    /// Default constructor. (Parameters)
+    /**
+     * @brief Construct a new Compute Nodal Gradient Process
+     * @details This constructor considers the model and retrieves the model parts
+     * @param rModel The model containing the model part
+     * @param ThisParameters User-defined parameters to construct the class
+     */
+    ComputeNodalGradientProcess(
+        Model& rModel,
+        Parameters ThisParameters = Parameters(R"({})")
+        );
+
+    /**
+     * @brief Construct a new Compute Nodal Gradient Process
+     * @details This constructor considers the model part as input
+     * @param rModelPart The model part containing the nodes and elements
+     * @param ThisParameters User-defined parameters to construct the class
+     */
     ComputeNodalGradientProcess(
         ModelPart& rModelPart,
         Parameters ThisParameters = Parameters(R"({})")
         );
 
-    /// Default constructor. (double)
+    /**
+     * @brief Computes the nodal gradient in a given ModelPart.
+     * @details This function computes the nodal gradient of a specified origin variable in the given ModelPart.
+     * The gradient is computed and stored in the specified gradient variable for each node.
+     * Optionally, the nodal area variable can be provided for weighted gradient computation.
+     * @param rModelPart The ModelPart in which the nodal gradient will be computed.
+     * @param rOriginVariable (optional) The variable for which the gradient will be computed.
+     * @param rGradientVariable (optional) The variable to store the computed gradient.
+     * @param rAreaVariable (optional) The variable representing nodal area for weighted gradient computation. Default is NODAL_AREA.
+     * @param NonHistoricalVariable (optional) Flag indicating if the origin variable is non-historical. Default is false.
+     */
     ComputeNodalGradientProcess(
         ModelPart& rModelPart,
-        const Variable<double>& rOriginVariable,
-        const Variable<array_1d<double,3> >& rGradientVariable,
+        const Variable<double>& rOriginVariable = DISTANCE,
+        const Variable<array_1d<double,3>>& rGradientVariable = DISTANCE_GRADIENT,
```

Comment:
Honestly I dont see why DISTANCE should be assumed to be the default variables

Reply:
No one is forced to use it, but we ca simply removed if you want